### PR TITLE
Add backward compatibility to last password update claim

### PR DESCRIPTION
**Pull request by ChanikaRuchini**
## Purpose
Resolves wso2/product-is#12690

- Added backward compatibility to last password update time claim in Password Policy Authenticator.

### CODE REVIEW

File: component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordChangeHandler.java [hunk 1]
```diff
@@ -18,8 +18,11 @@
 
 package org.wso2.carbon.identity.policy.password;
 
+import org.apache.commons.lang3.StringUtils;
```
Comment (pamodaaw) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordChangeHandler.java:21]: shouldn't we use `org.apache.commons.lang.StringUtils`

File: component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordResetEnforcer.java [hunk 2]
```diff
@@ -400,4 +398,21 @@ private void updateAuthenticatedUserInStepConfig(AuthenticationContext context,
...
         context.setSubject(authenticatedUser);
     }
+
+    private String getLastPasswordUpdateTime(UserStoreManager userStoreManager, String tenantAwareUsername)
+            throws org.wso2.carbon.user.core.UserStoreException {
+
+        Claim[] claims = userStoreManager.getUserClaimValues(tenantAwareUsername, null);
+        if (claims != null) {
+            for (Claim claim : claims) {
+                if (PasswordPolicyConstants.LAST_CREDENTIAL_UPDATE_TIMESTAMP_CLAIM.equals(claim.getClaimUri()) ||
```
Comment (pamodaaw) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordResetEnforcer.java:408]: Can we guarantee that the order of claims? What if the non-identity claim comes first in the array for the migrated users?

File: component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordResetEnforcer.java [hunk 3]
```diff
@@ -400,4 +419,17 @@ private void updateAuthenticatedUserInStepConfig(AuthenticationContext context,
         }
         context.setSubject(authenticatedUser);
     }
+
+    private String getLastPasswordUpdateTime(UserStoreManager userStoreManager, String claimURI,
+                                             String tenantAwareUsername)
+            throws UserStoreException {
```
Comment (pamodaaw) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordResetEnforcer.java:425]: check the formatting

Comment (ChanikaRuchini) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordResetEnforcer.java]: Fixed

**Review by pamodaaw — APPROVED** [review]

**Pull request merged**